visualize-admin/visualization-tool
# feat: Automatic sort based on identifier and position

Added a new sort option called "Automatic" for column / lines / area and pie chart. It uses either
the identifier or the position of the value to sort the values.
Legend is sorted according to this and also the tooltip content values.

Technically, I've tried to group the sorting logic that was dispersed
across chart to be able to more easily add the new sorting method.

- [ ] ~~Add alternate name as a sorting option~~ Will do that in another PR

## How to test

- Go to the preview deployment (See "Visit preview" link below)
- Go to the photovoltaik dataset *on int*: `/en/create/new?cube=https://energy.ld.admin.ch/sfoe/bfe_ogd84_einmalverguetung_fuer_photovoltaikanlagen/13&dataSource=Int`
- Select {areas,pie,column,lines} charts
- Go the segment and pick "Standort der Anlage" to color
- "Automatic" should be the default selected options, Zurich should come first, Jura should come last
- This should be visible in the chart, the legend, and the tooltip options
- You have the ability to revert the sort
- You should be able to select "Name" as a sorting option

Original issue : https://gitlab.ldbar.ch/bafu/umweltdatenkiosk-planning/-/issues/119

Fixes https://github.com/visualize-admin/visualization-tool/issues/789
